src/charm.py

```diff
@@ -368,6 +388,26 @@ def _on_collect_unit_status(self, e: CollectStatusEvent):
             else:
                 e.add_status(ActiveStatus())
 
+    def _on_nginx_pebble_ready(self, _) -> None:
+        self.nginx.configure_pebble_layer(tls=self.tls_available)
```

Comment:
instead of propagating `self.tls_available` down to where its needed, can we make, for example, a new property in `nginx.py` `tls_ready` that checks if certs are already present in the disk or not?

Reply:
True, that looks much better :)